elasticgraph-graphql/spec/integration/elastic_graph/graphql/datastore_query/filtering_spec.rb

```diff
@@ -1007,6 +1007,154 @@ def search_datastore(**options, &before_msearch)
         expect(results).to match_array(ids_of(widget1))
       end
 
+      specify "`matches_query_with_prefix` performs bool prefix matching with exact term matching" do
+        index_into(
+          graphql,
+          widget1 = build(:widget, description_text: "red widget small"),
+          widget2 = build(:widget, description_text: "red widget medium")
```

Comment:
It's important that these tests index some records which are not matched. Otherwise, an implementation that matches all documents (e.g. by leaving the query unmodified) can make the test pass!

In this case, it would be nice to show multiple examples of it not matching, including a case where a different `allowed_edits_per_term` value would make it match.